apps/opik-backend/src/main/java/com/comet/opik/domain/ProjectService.java

```diff
@@ -199,8 +202,19 @@ public Project get(@NonNull UUID id, @NonNull String workspaceId) {
                 .nonTransaction(connection -> traceDAO.getLastUpdatedTraceAt(Set.of(id), workspaceId, connection))
                 .block();
 
+        Map<UUID, Map<String, Object>> projectStats = getProjectStats(List.of(id), workspaceId);
+
+        return enhanceProject(project, lastUpdatedTraceAt, projectStats);
+    }
+
+    private Project enhanceProject(Project project, Map<UUID, Instant> lastUpdatedTraceAt,
+            Map<UUID, Map<String, Object>> projectStats) {
         return project.toBuilder()
-                .lastUpdatedTraceAt(lastUpdatedTraceAt.get(id))
+                .lastUpdatedTraceAt(lastUpdatedTraceAt.get(project.id()))
+                .feedbackScores(StatsMapper.getStatsFeedbackScores(projectStats.get(project.id())))
+                .duration(StatsMapper.getStatsDuration(projectStats.get(project.id())))
+                .totalEstimatedCost(StatsMapper.getStatsTotalEstimatedCost(projectStats.get(project.id())))
+                .usage(StatsMapper.getStatsUsage(projectStats.get(project.id())))
```

Comment:
Minor: it seems that `(projectStats.get(project.id())` should have been resolved only once. Probably externally to the method and passed as argument.